src/System.IO.Pipelines/tests/PipeLengthTests.cs

```diff
@@ -197,5 +197,15 @@ public async Task ExaminedLessThanBeforeThrows()
             result = await _pipe.Reader.ReadAsync();
             Assert.Throws<InvalidOperationException>(() => _pipe.Reader.AdvanceTo(result.Buffer.Start, result.Buffer.Start));
         }
+
+        [Fact]
+        public async Task ConsumedGreatherThanExaminedThrows()
+        {
+            _pipe.Writer.WriteEmpty(10);
+            await _pipe.Writer.FlushAsync();
+
+            ReadResult result = await _pipe.Reader.ReadAsync();
+            Assert.Throws<InvalidOperationException>(() => _pipe.Reader.AdvanceTo(result.Buffer.End, result.Buffer.Start));
```

Comment:
Should there be additional tests around the cases of consumed/examined being null (e.g. the typo highlighted earlier)?

Reply:
Added some